app/src/main/java/com/danielpriddle/drpnews/ui/ArticleListComposable.kt

```diff
@@ -0,0 +1,101 @@
+package com.danielpriddle.drpnews.ui
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Card
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import com.danielpriddle.drpnews.data.models.Article
+import androidx.compose.material.Text
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.danielpriddle.drpnews.data.models.Source
+import com.danielpriddle.drpnews.utils.State
+import com.danielpriddle.drpnews.viewmodels.ArticleListViewModel
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+
+@Composable
+fun ArticleListComposable(
+    articles: List<Article>,
+    clickListener: (Article) -> Unit = {},
+) {
+    val articleViewModel: ArticleListViewModel = viewModel()
+    SwipeRefresh(
+        state = rememberSwipeRefreshState(isRefreshing = articleViewModel.state.value == State.Loading),
```

Comment:
minor, but I typically like to set my remembered objects as variables at the top of my composable functions.  So, create a `val swipeRefreshState by rememberSwipRefreshState(isRefreshing = articleViewModel.state.value == State.Loading)` and then here:

```suggestion
        state = swipeRefreshState,
```